src/OrchardCore.Modules/OrchardCore.Flows/Drivers/BagPartDisplay.cs

```diff
@@ -34,13 +34,15 @@ IServiceProvider serviceProvider
 
         public override IDisplayResult Display(BagPart bagPart, BuildPartDisplayContext context)
         {
-            return Shape<BagPartViewModel>("BagPart", m =>
+            var hasItems = bagPart.ContentItems.Any();
+
+            return Shape<BagPartViewModel>(hasItems ? "BagPart" : "BagPart_NoItems", m =>
```

Comment:
`Empty` instead of `NoItems` ?

Reply:
I had used `Empty` to start with and then changed to `NoItems` last minute. Think I prefer `Empty` now that I've slept on it.